internal/bucket/bucket.go

```diff
@@ -24,10 +25,19 @@ var (
 	CID -> reference to content (map[string]interface{})
 */
 type BucketContent struct {
-	Item        interface{}
-	Id          string
-	ContentType bt.ResourceIdentifier
+	Item        interface{}           `json:"item"`
+	Id          string                `json:"id"`
+	ContentType bt.ResourceIdentifier `json:"contentType"`
 }
+
+func (bc *BucketContent) MarshalJson() ([]byte, error) {
+	return json.Marshal(bc)
```

Comment:
To implement `Marshaler` the function should be `MarshalJSON`. With the json tags above however, it should be unnecessary to implement. Instead you can do `json.Marshal(...)` in the calling code.

Reply:
right, i added this for ease when serializing for protobuf and other requests that might need the data in such a format. it was brought in to make it easier on the consumer. ill make the changes to 'marshalJSON' and 'unmarshalJSON'